viewer/src/tools/AxisView/AxisViewTool.ts

```diff
@@ -0,0 +1,321 @@
+  private readonly _layoutConfig: AxisBoxConfig;
+
+  private readonly _boxFaceGeometry: THREE.PlaneGeometry;
+
+  private readonly _viewer: Cognite3DViewer;
+
+  private readonly _interactiveObjects: THREE.Mesh[];
+  private readonly _raycastCamera: THREE.OrthographicCamera;
+  private readonly _raycaster: THREE.Raycaster;
+
+  private _screenPosition: THREE.Vector2;
+
+  private dynamicUpdatePosition = () => {};
+
+  constructor(viewer: Cognite3DViewer, config?: AxisBoxConfig) {
+    super();
+
+    this._screenPosition = new THREE.Vector2();
+    this._boxFaceGeometry = new THREE.PlaneGeometry(0.85, 0.85, 1, 1);
+
+    this._raycastCamera = new THREE.OrthographicCamera(-1, 1, 1, -1, -1, 1);
+    this._raycaster = new THREE.Raycaster();
+
+    this._viewer = viewer;
+
+    this._layoutConfig = _.merge(_.cloneDeep(defaultAxisBoxConfig), config);
+
+    const axisGroup = new THREE.Group();
+    this._interactiveObjects = this.createAxisCross(axisGroup);
+
+    this.addAxisBoxToViewer(axisGroup, this._layoutConfig.position!);
+  }
+
+  private addAxisBoxToViewer(axisGroup: THREE.Group, position: Absolute | Relative) {
+    const size = this._layoutConfig.size!;
+    this._viewer.addUiObject(axisGroup, this._screenPosition, size, size);
+
+    if (isAbsolute(position)) {
+      this._screenPosition.x = position.xAbsolute;
+      this._screenPosition.y = position.yAbsolute;
+    } else {
+      switch (position.corner) {
+        case Corner.BottomRight:
+          this._screenPosition.y = position.padding.y;
+
+          this.dynamicUpdatePosition = () => {
+            this._screenPosition.x = this._viewer.renderer.domElement.clientWidth - position.padding.x - size;
+          };
+          break;
+
+        case Corner.TopRight:
+          this.dynamicUpdatePosition = () => {
+            this._screenPosition.x = this._viewer.renderer.domElement.clientWidth - position.padding.x - size;
+            this._screenPosition.y = this._viewer.renderer.domElement.clientHeight - position.padding.y - size;
+          };
+          break;
+
+        case Corner.TopLeft:
+          this._screenPosition.x = position.padding.x;
+          this.dynamicUpdatePosition = () => {
+            this._screenPosition.y = this._viewer.renderer.domElement.clientHeight - position.padding.y - size;
+          };
+          break;
+
+        case Corner.BottomLeft:
+          this._screenPosition.x = position.padding.x;
+          this._screenPosition.y = position.padding.y;
+          break;
+        default:
+          throw new Error(`Unknown corner position for Axis Cross: Corner = ${position.corner}`);
+      }
+    }
+
+    function isAbsolute(position: Absolute | Relative): position is Absolute {
+      return (position as Absolute).xAbsolute !== undefined && (position as Absolute).yAbsolute !== undefined;
+    }
+  }
+
+  public tryClick(xScreenPos: number, yScreenPos: number, moveCameraOnSuccess = true): boolean {
```

Comment:
I think this can (and should) be avoided by listening to click/pointer events on `viewer.domElement`. When the axis tool is clicked, use `event.preventDefault()` to avoid that the viewer handles the click. By doing this, users would not have to call this function.